### DISCUSSION TITLE
BT-14-Lot

### DISCUSSION THREAD

**Discussion by elerik** [OP-TED/eForms-SDK#372] [Q&A] [answered]
For notice subtype 16 its mandatory. 

BT-14-LOT 
`"xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms/cac:CallForTendersDocumentReference/cbc:DocumentType"`

I am using CVS for validation. So i send xml and it return error message for that field

error message: 
`{"line":107,"column":27,"message":"cvc-complex-type.2.4.a: Invalid content was found starting with element '{\"urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2\":DocumentType}'. One of '{\"urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2\":UBLExtensions, \"urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2\":ID}' is expected."}`

so i found ID. Field. OPT-140-Lot is that ID which is missing. When i fill it. Error is gone. 
`"xpathAbsolute" : "/*/cac:ProcurementProjectLot[cbc:ID/@schemeName='Lot']/cac:TenderingTerms/cac:CallForTendersDocumentReference/cbc:ID",`
    
But this field is not mandatory for noticeSubtype 16 and I dont see anything what i could use to recognize that it has to be filled.
Am I missing something somewhere how to connect this 2 fields ? 

When i Export xml from https://enotices2.preview.ted.europa.eu/ I see this in xml. But I filled only BT-14-Lot (DocumentType)
`<cac:CallForTendersDocumentReference>`
`    <cbc:ID>default-text</cbc:ID>`
`    <cbc:DocumentType>non-restricted-document</cbc:DocumentType>`
` </cac:CallForTendersDocumentReference>`

So my question is. How do i know that I have to fill OPT-140-Lot (Id) because its requested in validation process. Or at least inject "default-text" even its not filled by user ?

**Comment by YvesJo** [comment, marked as answer]
Many elements appear as optional because of inactive conditional rules and it is the case for multiple Business terms of the Business Group BG-101 Procurement Documents. We are doing our best to have these rules active.
When stating only that procurement documents do not have restricted accesses, this does not provide much information and further information would be expected, like the URL where these documents may be found.
OPT-140-Lot is an Identifier mandated by the schema. It is required as soon as any information about the procurement documents is provided. Conditionally forbidden and mandatory rules have been defined for that. They will be activated with SDK 1.8.

**Reply by elerik** [reply]
So if i understand right 
From next update CVS in this situation will not answer as "xml is not valid against xsd" but will answer something more user friendly (like in other cases) for example "BT-21-Lot is mandatory for a notice with subtype 29 cbc:Name" ?

**Reply by YvesJo** [reply]
As long as the XML instance does not comply with the schema, the same answer will be provided. The conditional rule may be helpful when integrated in the UI as it will help the user detect the field is mandatory.
